### PR TITLE
[MASTERDIET-35] Activity search view

### PR DESCRIPTION
**Pull request by RafisSomeone**

### CODE REVIEW

File: src/app/App.js [hunk 1]
```diff
@@ -89,6 +90,11 @@ class App extends Component {
               currentUser={this.state.currentUser}
               component={ProductBrowser}
             />
+            <PrivateRoute
+              path='/activityBrowser' authenticated={this.state.authenticated}
```
Comment (jakubowiczish) [inline src/app/App.js:94]: Trivialism, but it's cool to have every props in new line 😄 but it's your choice, optional for me

File: src/productBrowser/RecentActivitiesComponent.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    this.state = {
+      activities: [],
+      pageIndex: 0,
+      maxPageNumber: 0
+    }
+    this.handleChangePage = this.handleChangePage.bind(this)
+    this.loadRecentProducts()
+  }
+
+  loadRecentProducts () {
```
Comment (jakubowiczish) [inline src/productBrowser/RecentActivitiesComponent.js:19]: These two methods are almost the same, I would refactor it somehow

File: src/productBrowser/SearchActivityComponent.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    this.handleSearchButtonClick = this.handleSearchButtonClick.bind(this)
+    this.handleSearchTermChange = this.handleSearchTermChange.bind(this)
+    this.handleChangePage = this.handleChangePage.bind(this)
+  }
+
+  handleSearchTermChange (newTerm) {
+    this.setState({ searchTerm: newTerm.target.value })
+  }
+
+  handleSearchButtonClick () {
```
Comment (jakubowiczish) [inline src/productBrowser/SearchActivityComponent.js:29]: Same in here - these two methods are almost the same, to me it needs refactoring

File: src/productBrowser/RecentActivitiesComponent.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+class RecentActivitiesComponent extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      activities: [],
+      pageIndex: 0,
+      maxPageNumber: 0
+    }
+    this.handleChangePage = this.handleChangePage.bind(this)
+    this.loadRecentProducts()
```
Comment (jakubowiczish) [inline src/productBrowser/RecentActivitiesComponent.js:16]: I believe it should be done in ```componentDidMount```

File: src/productBrowser/ActivityItemComponent.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React, { Component } from 'react'
+import './ProductBrowser.css'
+
+class ActivityItemComponent extends Component {
```
[the remark this answers is not in the document]
Comment (jakubowiczish) [reply src/productBrowser/ActivityItemComponent.js:4]: To be updated - new componend made by @jfoder

[the remark this answers is not in the document]
Comment (RafisSomeone) [reply src/productBrowser/ActivityItemComponent.js:4]: changed to @jfoder component

File: src/app/App.js (hunk 1 above)
Comment (RafisSomeone) [reply src/app/App.js:95]: changed

File: src/productBrowser/RecentActivitiesComponent.js (hunk 2 above)
Comment (RafisSomeone) [reply src/productBrowser/RecentActivitiesComponent.js:24]: changed to new component

File: src/productBrowser/SearchActivityComponent.js (hunk 3 above)
Comment (RafisSomeone) [reply src/productBrowser/SearchActivityComponent.js:33]: changed to new component

File: src/productBrowser/RecentActivitiesComponent.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+      {
+        name: 'Name',
+        selector: 'name',
+        sortable: true,
+        grow: 2.5,
+        wrap: true,
+        center: true
+      },
+      {
+        name: 'Default portion',
```
Comment (jfoder) [inline src/productBrowser/RecentActivitiesComponent.js:77]: I think this column should be renamed to something like `Default reps` or `Default sets`

File: src/productBrowser/RecentActivitiesComponent.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+      perPage: 10
+    }
+    this.getColumnsForRecentActivitiesComponent = this.getColumnsForRecentActivitiesComponent.bind(this)
+    this.handleSelectChange = this.handleSelectChange.bind(this)
+    this.handlePageChange = this.handlePageChange.bind(this)
+    this.handlePerRowsChange = this.handlePerRowsChange.bind(this)
+    this.loadRecentProducts()
+  }
+
+  loadRecentProducts () {
```
Comment (jfoder) [inline src/productBrowser/RecentActivitiesComponent.js:24]: It should be renamed to something like `loadRecentActivities`

File: src/productBrowser/SearchActivityComponent.js [hunk 8]
```diff
@@ -0,0 +1,155 @@
+import React, { Component } from 'react'
+import InputGroup from 'react-bootstrap/InputGroup'
+import FormControl from 'react-bootstrap/FormControl'
+import Button from 'react-bootstrap/Button'
+import './ProductBrowser.css'
```
Comment (jfoder) [inline src/productBrowser/SearchActivityComponent.js:5]: This css file is created for product browser. Maybe we should create some common css file for search components?

File: src/productBrowser/RecentActivitiesComponent.js (hunk 6 above)
Comment (RafisSomeone) [reply src/productBrowser/RecentActivitiesComponent.js:77]: It's unnecessary. Deleted.

File: src/productBrowser/RecentActivitiesComponent.js (hunk 7 above)
Comment (RafisSomeone) [reply src/productBrowser/RecentActivitiesComponent.js:24]: Done.

File: src/productBrowser/RecentActivitiesComponent.js (hunk 6 above)
Comment (RafisSomeone) [reply src/productBrowser/RecentActivitiesComponent.js:77]: Change to the global search class.

File: src/productBrowser/SearchActivityComponent.js (hunk 8 above)
Comment (RafisSomeone) [reply src/productBrowser/SearchActivityComponent.js:5]: Changed to the global searched class

**Pull request merged**